code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm

```diff
@@ -100,11 +118,16 @@
 
 /obj/machinery/atmospherics/components/binary/thermomachine/update_overlays()
 	. = ..()
-	. += getpipeimage(icon, "pipe", dir, COLOR_LIME, piping_layer)
-	. += getpipeimage(icon, "pipe", turn(dir, 180), COLOR_MOSTLY_PURE_RED, piping_layer)
+	if(!initial(icon))
+		return
+	var/mutable_appearance/thermo_overlay = new(initial(icon))
+	. += getpipeimage(thermo_overlay, "pipe", dir, COLOR_LIME, piping_layer)
+	. += getpipeimage(thermo_overlay, "pipe", turn(dir, 180), COLOR_MOSTLY_PURE_RED, piping_layer)
 	if(skipping_work && on)
-		var/mutable_appearance/skipping = mutable_appearance(icon, "blinking")
-		. += skipping
+		thermo_overlay.icon_state = "blinking"
+		. += new /mutable_appearance(thermo_overlay)
+		//var/mutable_appearance/skipping = mutable_appearance(thermo_overlay, "blinking")
+		//. += skipping
```

Comment:
Did you intend to leave this in?

Reply:
Missed this when I was cleaning up code.  Fixed with 5780de90a09af1ff23140910f23c2ae4155d7257